components/time/src/provider/mod.rs

```diff
@@ -215,14 +217,199 @@ impl<'de> serde::Deserialize<'de> for VariantOffsets {
     }
 }
 
-icu_provider::data_marker!(
-    /// The default mapping between period and offsets. The second level key is a wall-clock time encoded as
-    /// [`ZoneNameTimestamp`]. It represents when the offsets started to be used.
+/// Metazone ID in a compact format
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+pub type MetazoneId = core::num::NonZeroU8;
+
+/// Data struct for the [`TimezonePeriodsV1`] marker.
+#[derive(PartialEq, Debug, Clone, yoke::Yokeable, zerofrom::ZeroFrom)]
+#[cfg_attr(feature = "datagen", derive(databake::Bake))]
+#[cfg_attr(feature = "datagen", databake(path = icu_time::provider))]
+pub struct TimezonePeriods<'a> {
+    /// Index of `TimeZone`s into `list`.
+    pub index: ZeroTrieSimpleAscii<ZeroVec<'a, u8>>,
+    /// Each value contains at least one period, which implicitly starts at the UNIX epoch.
+    /// This is stored in the first tuple element.
+    ///
+    /// If more periods are required the second tuple element contains them, along with their
+    /// starting timestamp. These entries are ordered chronologically.
     ///
-    /// The values are the standard offset, and the daylight offset *relative to the standard offset*. As such,
-    /// if the second value is 0, there is no daylight time.
-    TimezoneVariantsOffsetsV1,
-    "timezone/variants/offsets/v1",
-    ZeroMap2d<'static, TimeZone, ZoneNameTimestamp, VariantOffsets>,
-    is_singleton = true
+    /// The values ((u8, Option<MetazoneId)) are an index into the offsets list for the offset
+    /// that the zone observes in that period, and optionally whether it is part of a metazone.
+    pub list: VarZeroVec<
+        'a,
+        VarTupleULE<
+            (u8, NichedOption<MetazoneId, 1>),
+            ZeroSlice<(Timestamp24, u8, NichedOption<MetazoneId, 1>)>,
+        >,
+    >,
+
+    /// The deduplicated list of offsets.
+    ///
+    /// There are currently 83 unique VariantOffsets, so storing the index as a u8 is plenty enough.
+    pub offsets: ZeroVec<'a, VariantOffsets>,
+}
+
+/// Encodes ZoneNameTimestamp in 3 bytes by dropping the unused metadata
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
+pub struct Timestamp24(pub ZoneNameTimestamp);
+
+impl AsULE for Timestamp24 {
+    type ULE = RawBytesULE<3>;
+    #[inline]
+    fn to_unaligned(self) -> Self::ULE {
+        let RawBytesULE([a, b, c, _]) = self.0.to_unaligned();
```

Comment:
Thought: I had included the metadata in the ZoneNameTimestamp in case we needed it to encode the offset at which the timestamp occurred. Now that we are always storing it in UTC, we don't need the metadata right now, correct?